### PR TITLE
[docs] add doc for datadogdashboard

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Add documentation for DatadogDashboard.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/datadog_dashboard.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+- Datadog Operator v1.9+
+- **[Helm][2]** for deploying the Datadog Operator
+- **[`kubectl` CLI][3]** for installing a `DatadogDashboard`
+
+Tests were performed on Kubernetes versions >= `v1.30.0`
+
+## Adding a DatadogDashboard
+
+To deploy a `DatadogDashboard` with the Datadog Operator, use the [`datadog-operator` Helm chart][3].
```
Comment (levan-m) [inline docs/datadog_dashboard.md:17]:
```suggestion
To deploy a `DatadogDashboard` with the Datadog Operator, use the [`datadog-operator` Helm chart][4].
```

Comment (levan-m) [reply docs/datadog_dashboard.md:17]: Not sure if you looked but rendering hyperlink with code and text mix looks kind of broken 
  
Screenshot 2024-09-30 at 11 28 50 AM

File: docs/datadog_dashboard.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+1. Install the [Datadog Operator][4]:
+
+   First, add the Datadog Helm chart with
+
+    ```shell
+    helm repo add datadog https://helm.datadoghq.com
+    ```
+
+    1. Run the following command, substituting your [Datadog API and application keys][5]:
```
Comment (levan-m) [inline docs/datadog_dashboard.md:27]: [5] takes you to https://app.datadoghq.com/dashboard/lists not the API/APP keys or public doc were to find those

**Review by levan-m — APPROVED** [review]

File: docs/datadog_dashboard.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
+# Datadog Dashboards
+This feature is in preview.
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:2]:
```suggestion
This feature is in Preview.
```

File: docs/datadog_dashboard.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+## Overview
+The DatadogDashboard Custom Resource Definition (CRD) allows users to create [dashboards][1] using the Operator and manage them as Kubernetes resources.
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:5]:
```suggestion
The `DatadogDashboard` Custom Resource Definition (CRD) allows users to create [dashboards][1] using the Operator and manage them as Kubernetes resources.
```

File: docs/datadog_dashboard.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+## Prerequisites
+
+- Datadog Operator v1.9+
+- **[Helm][2]** for deploying the Datadog Operator
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:10]:
```suggestion
- [Helm][2], to deploy the Datadog Operator
```

File: docs/datadog_dashboard.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+- **[`kubectl` CLI][3]** for installing a `DatadogDashboard`
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:11]:
```suggestion
- The [`kubectl` CLI][3], to install a `DatadogDashboard`
```

File: docs/datadog_dashboard.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+Tests were performed on Kubernetes versions >= `v1.30.0`
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:13]: Is this line necessary? It seems to me like this should either be a prereq or shouldn't be mentioned.

File: docs/datadog_dashboard.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        ```shell
+        helm install my-datadog-operator datadog/datadog-operator --set apiKey=<DATADOG_API_KEY> --set appKey=<DATADOG_APP_KEY> --set datadogDashboard.enabled=true
+        ```
+
+    1. Alternatively, update the [`values.yaml`][7] file of the Datadog Operator Helm chart to include your [Datadog API and application keys][6] and enable `DatadogDashboard`.
+       Then, run
+
+        ```shell
+        helm install my-datadog-operator datadog/datadog-operator -f values.yaml
+        ```
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:38]:
```suggestion
1. To install the [Datadog Operator][5], first add the Datadog Helm chart using the following command:
    ```shell
    helm repo add datadog https://helm.datadoghq.com
    ```
1. Choose one of the following options:
    * Run the following command, substituting your [Datadog API and application keys][6]:

        ```shell
        helm install my-datadog-operator datadog/datadog-operator --set apiKey=<DATADOG_API_KEY> --set appKey=<DATADOG_APP_KEY> --set datadogDashboard.enabled=true
        ```

    * Update the [`values.yaml`][7] file of the Datadog Operator Helm chart to include your [Datadog API and application keys][6] and enable `DatadogDashboard`, then run the following command:

        ```shell
        helm install my-datadog-operator datadog/datadog-operator -f values.yaml
        ```
```

File: docs/datadog_dashboard.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                ]
+            },
+            "layout": {
+                "x": 0,
+                "y": 0,
+                "width": 4,
+                "height": 2
+            }
+         }]'
+```
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:110]:
~~~suggestion
    ```
    apiVersion: datadoghq.com/v1alpha1
    kind: DatadogDashboard
    metadata:
    name: example-dashboard
    spec:
    title: Test Dashboard
    layoutType: ordered
    tags:
        - "team:my_team"
    templateVariables:
        - availableValues: 
            - host1
            - host2
            - host3
        name: first
        prefix: bar-foo
    notifyList:
        - foobar@example.com
    widgets: '[{
                "id": 2639892738901474,
                "definition": {
                    "title": "",
                    "title_size": "16",
                    "title_align": "left",
                    "show_legend": true,
                    "legend_layout": "auto",
                    "legend_columns": [
                        "avg",
                        "min",
                        "max",
                        "value",
                        "sum"
                    ],
                    "type": "timeseries",
                    "requests": [
                        {
                            "formulas": [
                                {
                                    "formula": "query1"
                                }
                            ],
                            "queries": [
                                {
                                    "name": "query1",
                                    "data_source": "metrics",
                                    "query": "avg:system.cpu.user{*} by {host}"
                                }
                            ],
                            "response_format": "timeseries",
                            "style": {
                                "palette": "dog_classic",
                                "order_by": "values",
                                "line_type": "solid",
                                "line_width": "normal"
                            },
                            "display_type": "line"
                        }
                    ]
                },
                "layout": {
                    "x": 0,
                    "y": 0,
                    "width": 4,
                    "height": 2
                }
            }]'
    ```
~~~

File: docs/datadog_dashboard.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    ```shell
+    kubectl apply -f /path/to/your/datadog-dashboard.yaml
+    ```
+
+    This automatically creates a new dashboard in Datadog. You can find it on the [Dashboards][8] page of your Datadog account.
+
+
+## Cleanup
+
+The following commands delete the dashboard from your Datadog account and all the Kubernetes resources created by the above instructions:
```
Comment (neko-dd) [inline docs/datadog_dashboard.md:123]:
```suggestion
The following commands delete the dashboard from your Datadog account as well as all of the Kubernetes resources created by the previous instructions:
```

**Pull request merged**